plugin/aws/ec2Instance.go

```diff
@@ -238,457 +191,13 @@ func (inst *ec2Instance) checkLatestConsoleOutput(ctx context.Context) (*ec2Inst
-	commandID string,
-	instanceID string,
-) *outputLogStream {
-	return &outputLogStream{
-		name:         name,
-		created:      false,
-		nextToken:    "",
-		client:       client,
-		logGroupName: "/aws/ssm/AWS-RunShellScript",
-		logStreamName: fmt.Sprintf(
-			"%v/%v/aws-runShellScript/%v",
-			commandID,
-			instanceID,
-			name,
-		),
-	}
-}
-
-func (s *outputLogStream) nextEvents(ctx context.Context) ([]*cloudwatchlogs.OutputLogEvent, error) {
-	activity.Record(
-		ctx,
-		"Fetching the next CloudWatch log events for %v. Log group name: %v. Log stream name: %v",
-		s.name,
-		s.logGroupName,
-		s.logStreamName,
-	)
-
-	request := &cloudwatchlogs.GetLogEventsInput{
-		LogGroupName:  awsSDK.String(s.logGroupName),
-		LogStreamName: awsSDK.String(s.logStreamName),
-	}
-	if s.nextToken == "" {
-		// This is our first request
-		request.StartFromHead = awsSDK.Bool(true)
-	} else {
-		request.NextToken = awsSDK.String(s.nextToken)
-	}
-
-	resp, err := s.client.GetLogEventsWithContext(ctx, request)
-	if err != nil {
-		if awserr, ok := err.(awserr.Error); ok {
-			if awserr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
-				if s.created {
-					return nil, fmt.Errorf("the CloudWatch logs for %v were unexpectedly deleted", s.name)
-				}
-
-				// The stream hasn't been created. This is OK. It means that
-				// AWS hasn't sent them over yet, OR the command hasn't printed
-				// anything to this stream.
-				activity.Record(
-					ctx,
-					"Did not fetch any events for %v: the log stream hasn't been created yet",
-					s.name,
-				)
-
-				return []*cloudwatchlogs.OutputLogEvent{}, nil
-			}
-		}
-
-		return nil, fmt.Errorf("failed to fetch the events for %v: %v", s.name, err)
-	}
-	s.created = true
-	s.nextToken = awsSDK.StringValue(resp.NextForwardToken)
-
-	activity.Record(ctx, "Fetched %v new events for %v", len(resp.Events), s.name)
-	return resp.Events, nil
-}
-
-func (s *outputLogStream) Close(ctx context.Context) {
-	if !s.created {
-		return
-	}
-
-	activity.Record(
-		ctx,
-		"Deleting the logs for %v. Log group name: %v. Log stream name: %v",
-		s.name,
-		s.logGroupName,
-		s.logStreamName,
-	)
-
-	request := &cloudwatchlogs.DeleteLogStreamInput{
-		LogGroupName:  awsSDK.String(s.logGroupName),
-		LogStreamName: awsSDK.String(s.logStreamName),
-	}
-
-	// Use a different context to delete the log stream so that Wash
-	// will still attempt to delete it when our current context is
-	// cancelled
-	deleteCtx, cancelFunc := context.WithTimeout(context.Background(), 3*time.Second)
-	defer cancelFunc()
-
-	// DeleteLogStream's response does not contain any useful information
-	// so we can discard it
-	_, err := s.client.DeleteLogStreamWithContext(deleteCtx, request)
-	if err != nil {
-		activity.Record(ctx, "Failed to delete the logs for %v: %v", s.name, err)
-	}
+	return transport.ExecSSH(ctx, transport.Identity{Host: dnsname, User: "ec2-user"}, append([]string{cmd}, args...), opts)
```

Comment:
How do we decide what user to connect as? New option for exec?

Ideally we'd have a way to determine the user we should connect as from metadata, but not always exposed. We need a default (or configurable) user for using `fs`.

Reply:
I think we could go with ec2-user based on what I'm seeing in https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AccessingInstancesLinux.html.

Reply:
ec2-user is only the default for Amazon AMIs. It's not a bad default, using ssh config to override it. If I can find an authoritative source I can query or hard-code I might make it smarter.